### PR TITLE
Merge custom overrides representation

### PR DESCRIPTION
**Pull request by Jhonnyg**
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: editor/src/clj/editor/graphics.clj [hunk 1]
```diff
@@ -270,22 +279,35 @@
...
+                        ;; properties may also produce a different element count from
+                        ;; what the material dictates.
+                        (let [resized-values (resize-doubles override-values semantic-type element-count)
+                              [attribute-value-keyword stored-values] (doubles->storage resized-values data-type normalize)]
+                          {:name name
+                           attribute-value-keyword {:v stored-values}}))))
+              material-attribute-infos)
+        orphaned-attribute-save-values
+        (into []
+              (keep (fn [attribute]
```
Comment (matgis) [inline editor/src/clj/editor/graphics.clj:300]: We can use destructuring on the argument here to make things clearer:
```clj
(fn [[name-key attribute-info]]
  ,,,)
```

File: editor/src/clj/editor/model.clj [hunk 2]
```diff
@@ -619,7 +611,7 @@
   (output animation-set-build-target-single g/Any :cached produce-animation-set-build-target-single)
 
   (output pb-msg g/Any :cached produce-pb-msg)
-  (output save-value g/Any (gu/passthrough pb-msg))
+  (output save-value g/Any produce-save-value)
```
Comment (matgis) [inline editor/src/clj/editor/model.clj:614]: I think we should add `:cached` on this `output` since it does quite a lot of work now?

**Review by matgis — CHANGES_REQUESTED** [review]
Nice job with using consistent terms everywhere! It's getting there.

I found some inefficiencies and things that could be improved, and I really think we need to `:cache` the `save-value` output after this change.

File: editor/src/clj/editor/graphics.clj [hunk 3]
```diff
@@ -319,7 +341,13 @@
     (:element-count attribute)))
 
 (defn- attribute-update-property [current-property-value attribute new-value]
-  (assoc current-property-value (:name-key attribute) new-value))
+  (let [override-info ((:name-key attribute) current-property-value)
+        attribute-value-keyword (attribute-value-keyword (:data-type attribute) (:normalize attribute))
+        override:value-keyword (or attribute-value-keyword (:value-keyword override-info))
```
[the remark this answers is not in the document]
Comment (Jhonnyg) [reply editor/src/clj/editor/graphics.clj:346]: Not sure I follow, are you talking about the (:name-key attribute) ? I need to get the already stored data and only overwrite it if we have valid data, so I need the name-key from the incoming attribute

[the remark this answers is not in the document]
Comment (matgis) [reply editor/src/clj/editor/graphics.clj:346]: I mean `override:value-keyword` is an unconventional name for a let-binding.

[the remark this answers is not in the document]
Comment (Jhonnyg) [reply editor/src/clj/editor/graphics.clj:346]: Screw convention, I like to stick it to the man in very small ways! (it was a typo)

File: editor/src/clj/editor/graphics.clj [hunk 4]
```diff
@@ -270,22 +279,35 @@
...
-                  ;; properties may also produce a different element count from
-                  ;; what the material dictates.
-                  (let [resized-values (resize-doubles override-values semantic-type element-count)
-                        [attribute-value-keyword stored-values] (doubles->storage resized-values data-type normalize)]
-                    {:name name
-                     attribute-value-keyword {:v stored-values}}))))
-        material-attribute-infos))
-
-(defn attributes->build-target [material-attribute-infos vertex-attribute-overrides vertex-attribute-bytes]
+(defn vertex-attribute-overrides->save-values [material-attribute-infos vertex-attribute-overrides]
```
[the remark this answers is not in the document]
Comment (Jhonnyg) [reply editor/src/clj/editor/graphics.clj:282]: Sure! I did the same for vertex-attribute-overrides->build-target

**Pull request merged**